tests/dataset/test_export_datasets_and_create_metadata_db.py

```diff
@@ -0,0 +1,477 @@
+    
+    # Check that target database was created
+    assert target_db_path.exists()
+    
+    # Check that target database has the run
+    target_conn = connect(target_db_path)
+    target_runs = get_runs(target_conn)
+    assert len(target_runs) == 1
+    target_conn.close()
+    
+    # Check that NetCDF file was created if export was successful
+    if result[run_id] == "exported":
+        netcdf_files = list(export_path.glob("*.nc"))
+        assert len(netcdf_files) > 0
+
+
+def test_export_datasets_preserve_experiment_structure():
+    """Test that experiment structure is preserved in the target database"""
+    with tempfile.TemporaryDirectory() as temp_dir:
+        source_db_path = Path(temp_dir) / "source.db"
+        target_db_path = Path(temp_dir) / "target.db"
+        export_path = Path(temp_dir) / "exports"
+        
+        # Create source database with multiple experiments
+        source_conn = connect(source_db_path)
+        
+        # Create first experiment
+        exp1 = load_or_create_experiment(
+            experiment_name="exp1",
+            sample_name="sample1",
+            conn=source_conn
+        )
+        
+        # Create second experiment
+        exp2 = load_or_create_experiment(
+            experiment_name="exp2",
+            sample_name="sample2",
+            conn=source_conn
+        )
+        
+        # Create interdependencies
+        x = ParamSpec("x", "numeric", unit="V")
+        y = ParamSpec("y", "numeric", unit="A")
+        interdeps = InterDependencies_(dependencies={y: (x,)})
+        
+        # Create datasets in both experiments
+        datasets = []
+        for exp in [exp1, exp2]:
+            for i in range(2):  # 2 datasets per experiment
+                dataset = DataSet(conn=source_conn, exp_id=exp.exp_id)
+                dataset.set_interdependencies(interdeps)
+                dataset.mark_started()
+                
+                # Add some data
+                for j in range(5):
+                    dataset.add_results([{"x": j, "y": j * (i + 1)}])
+                
+                dataset.mark_completed()
+                datasets.append(dataset)
+        
+        source_conn.close()
+        
+        # Run the export function
+        result = export_datasets_and_create_metadata_db(
+            source_db_path=source_db_path,
+            target_db_path=target_db_path,
+            export_path=export_path,
+        )
+        
+        # Check that all datasets were processed
+        assert len(result) == 4
+        
+        # Check that target database has all runs
+        target_conn = connect(target_db_path)
+        target_runs = get_runs(target_conn)
+        assert len(target_runs) == 4
+        target_conn.close()
+
+
+def test_export_datasets_with_incomplete_dataset():
+    """Test behavior when source database contains incomplete datasets"""
+    with tempfile.TemporaryDirectory() as temp_dir:
```

Comment:
please use tmp_path fixture instead of tempfile module, and make sure all the tests are update accordingly